### PR TITLE
[FUNCTIONNAL TEST] Login

### PR DESCRIPTION
**Pull request by cmarianhodis**
Related issue: https://github.com/backbee/backbee-standard/issues/590

### CODE REVIEW

File: nightwatch.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
 {
-  "src_folders" : ["nightwatch/tests"],
-  "output_folder" : "nightwatch/reports",
-  "custom_commands_path" : "",
-  "custom_assertions_path" : "",
-  "page_objects_path" : "",
-  "globals_path" : "nightwatch/configuration/globals.json",
+    "src_folders" : ["nightwatch/tests"],
+    "output_folder" : "nightwatch/reports",
+    "custom_commands_path" : ["nightwatch/commands/page", "nightwatch/commands/utils"],
```
Comment (fkroockmann) [inline nightwatch.json]: just nightwatch/commands works ?

File: nightwatch/commands/page/closeLoginPopin.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * @category    NightWatch
+ * @subcategory CustomCommands
+ * @copyright   Lp digital system
+ * @author      Marian Hodis <marian.hodis@lp-digital.fr>
+ */
+
+module.exports.command = function (callback) {
+    'use strict';
+    var self = this,
```
Comment (fkroockmann) [inline nightwatch/commands/page/closeLoginPopin.js]: space between 'use strict and var plz

File: nightwatch/tests/user/login.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+ *
+ * @category    NightWatch
+ * @subcategory Tests
+ * @copyright   Lp digital system
+ * @author      Marian Hodis <marian.hodis@lp-digital.fr>
+ */
+
+var openLoginPopin = function (client) {
+    'use strict';
+    client
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        .waitForElementPresent('body', client.globals.loadTime.body)
+        .openLoginPopin();
+};
+
+var fillLoginForm = function (client, loginSection, username, password) {
+    'use strict';
+    loginSection
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+module.exports = {
+    /**
+     * Set some usefull global variables
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    before : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    },
+
+    /**
+     * Open the baseUrl before each test
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    beforeEach : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+     * 
+     * Checks:
+     * 1. The login popin appears
+     * 2. When the login popin is closed the page refreshes
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    testOpenCloseLoginPopin : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+     * 
+     * Checks: 
+     * 1. The user is not logged in
+     * 2. The error message appears
+     * 
+     * @param {type} client
+     * @returns {undefined}
+     */
+    testLoginFakeUser : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+     * 1. The user is logged in
+     * 2. The page is refreshed
+     * 3. The toolbar is present
+     * 4. The user is logged out
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    testLoginLogoutRealUser : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+    },
+
+    /**
+     * After all tests are ran we end the client
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    after : function (client) {
+        'use strict';
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: space plz

File: nightwatch/tests/user/login.js [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+     * Test open / close login popin
+     * 
+     * Checks:
+     * 1. The login popin appears
+     * 2. When the login popin is closed the page refreshes
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    testOpenCloseLoginPopin : function (client) {
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: Can you use a literal variable for more explicit in console plz ?

File: nightwatch/tests/user/login.js [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+     * Test login with a fake user
+     * 
+     * Checks: 
+     * 1. The user is not logged in
+     * 2. The error message appears
+     * 
+     * @param {type} client
+     * @returns {undefined}
+     */
+    testLoginFakeUser : function (client) {
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: Can you use a literal variable for more explicit in console plz ?

File: nightwatch/tests/user/login.js [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+     * Checks:
+     * 1. The user is logged in
+     * 2. The page is refreshed
+     * 3. The toolbar is present
+     * 4. The user is logged out
+     * 
+     * @param {Object} client
+     * @returns {undefined}
+     */
+    testLoginLogoutRealUser : function (client) {
```
Comment (fkroockmann) [inline nightwatch/tests/user/login.js]: Can you use a literal variable for more explicit in console plz ?

File: nightwatch.json (hunk 1 above)
Comment (cmarianhodis) [inline nightwatch.json]: it doesn't take into account the files in subfolders, all folders need to be included

**Pull request merged**